File: Majority_Number.cpp
class Solution {
public:
    /**
     * @param nums: A list of integers
     * @return: The majority number
     */
    int majorityNumber(vector<int> nums) {
        // write your code here
        int ans = nums[0], cnt = 0;
        for (const auto& i : nums){
            if (ans == i){
                ++cnt;
            }else{
                --cnt;
                if (cnt == 0){
                    ans = i;
                    ++cnt;
                }
            }
        }
        return ans;
    }
};

File: Majority Number.py
class Solution:
    """
    @param nums: A list of integers
    @return: The majority number
    """
    def majorityNumber(self, nums):
        # write your code here
        
        count = 1
        ans = float('inf')
        for n in nums:
            if n != ans:
                count -= 1
                if count == 0:
                    count = 1
                    ans = n
            else:
                count += 1
        return ans
